include/zephyr/net/net_if.h

```diff
@@ -2535,6 +2535,13 @@ bool net_if_is_suspended(struct net_if *iface);
  */
 bool net_if_is_wifi(struct net_if *iface);
 
+/**
+ * @brief Get first Wi-Fi network interface.
+ *
+ * @return Pointer to network interface, NULL if not found.
+ */
+struct net_if *net_if_get_first_wifi(void);
```

Comment:
I think commits:
* `net: wifi: Add support to get and set Wi-Fi as default interface`
*  `net: wifi: Use only Wi-Fi interfaces`

could be a part of #58869 - they don't seem related to the rest of `wifi_mgmt` changes in here, and are a nice supplement to changes in #58869, as they actually show the reason for those changes.